src/ShearStressTransportEquationSystem.C

```diff
@@ -271,6 +276,49 @@ ShearStressTransportEquationSystem::solve_and_update()
     tkeEqSys_->compute_projected_nodal_gradient();
     sdrEqSys_->assemble_nodal_gradient();
   }
+
+}
+
+void clip_sst(
+  const stk::mesh::NgpMesh& ngpMesh,
+  const stk::mesh::Selector& sel,
+  const stk::mesh::NgpField<double>& density,
+  const stk::mesh::NgpField<double>& viscosity,
+  stk::mesh::NgpField<double>& tke,
+  stk::mesh::NgpField<double>& sdr)
+{
+  tke.sync_to_device();
+  sdr.sync_to_device();
+
+  const double clipValue = 1.0e-8;
+  nalu_ngp::run_entity_algorithm(
+    "SST::update_and_clip", ngpMesh, stk::topology::NODE_RANK, sel,
+    KOKKOS_LAMBDA(const nalu_ngp::NGPMeshTraits<>::MeshIndex& mi) {
+      const double tkeNew = tke.get(mi, 0);
+      const double sdrNew = sdr.get(mi, 0);
+
+      if ((tkeNew >= 0.0) && (sdrNew >= 0.0)) {
+        tke.get(mi, 0) = tkeNew;
+        sdr.get(mi, 0) = sdrNew;
+      } else if ((tkeNew < 0.0) && (sdrNew < 0.0)) {
+        // both negative; set TKE to small value, tvisc to molecular visc and use
+        // Prandtl/Kolm for SDR
+        tke.get(mi, 0) = clipValue;
+        sdr.get(mi, 0) = density.get(mi, 0) * clipValue / viscosity.get(mi, 0);
+      } else if (tkeNew < 0.0) {
+        // only TKE is off; reset turbulent viscosity to molecular vis and
+        // compute new TKE based on SDR and tvisc
+        sdr.get(mi, 0) = sdrNew;
+        tke.get(mi, 0) = viscosity.get(mi, 0) * sdrNew / density.get(mi, 0);
+      } else {
+        // Only SDR is off; reset turbulent viscosity to molecular visc and
+        // compute new SDR based on others
+        tke.get(mi, 0) = tkeNew;
+        sdr.get(mi, 0) = density.get(mi, 0) * tkeNew / viscosity.get(mi, 0);
+      }
+    });
+  tke.modify_on_device();
+  sdr.modify_on_device();
```

Comment:
remove this as this was recently addressed in #174 and af5102224fa71160

```suggestion
```

Reply:
👍